### PR TITLE
Use the same rocksdb snapshot when sending snapshot

### PR DESCRIPTION
**Pull request by critical27**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [X] enhancement

#### Description:

This is some prepare work for later data balance test, mainly two changes:
1. Follow up of #3717(In that pr we use snapshot when doing data scan, but the commit log id/term is not taken care). So in this PR, we create a rocksdb snapshot at first, read commit log id/term, and use the same snapshot to scan data.
2. We can't simply skip the special raft command, because if we skip the special log after rebooting, the state of raft peer will be wrong.
3. Add some metrics in raft, remove the `SlowOpTracker`.

## Checklist:
Tests:
- [X] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:
not related

### CODE REVIEW

**Review by SuperYoko — APPROVED** [review]

File: src/kvstore/NebulaSnapshotManager.cpp [hunk 1]
```diff
@@ -29,40 +29,69 @@ NebulaSnapshotManager::NebulaSnapshotManager(NebulaStore* kv) : store_(kv) {
...
-  LOG(INFO) << folly::sformat(
-      "Space {} Part {} start send snapshot, rate limited to {}, batch size is {}",
-      spaceId,
-      partId,
-      FLAGS_snapshot_part_rate_limit,
-      FLAGS_snapshot_batch_size);
-
-  const void* snapshot = store_->GetSnapshot(spaceId, partId);
+  CHECK_NOTNULL(store_);
+  // Get the commit log id and commit log term of specified partition
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/kvstore/NebulaSnapshotManager.cpp:38]: Nice catch

File: src/kvstore/Part.cpp [hunk 2]
```diff
@@ -308,26 +308,12 @@ std::tuple<nebula::cpp2::ErrorCode, LogID, TermID> Part::commitLogs(
       }
       case OP_TRANS_LEADER: {
         auto newLeader = decodeHost(OP_TRANS_LEADER, log);
-        auto ts = getTimestamp(log);
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/kvstore/Part.cpp:311]: It is like I said in the summary, there are some special logs for membership change, but these log is simply discarded when we reboot (by comparing the timestamp in the log and process start time). If the log is discarded, the raft peer could be wrong.

[the remark this answers is not in the document]
Comment (wenhaocs) [reply src/kvstore/Part.cpp:311]: make sense!

**Review by wenhaocs — APPROVED** [review]
LGTM. Want to make sure deleting SlowOpTracker is also intended.

**Comment by critical27** [comment]
> LGTM. Want to make sure deleting SlowOpTracker is also intended.

Yeah, `SlowOpTracker` is only used in raft, and after log level is adjusted, `SlowOpTracker` still use `LOG(INFO)` which is not appropriate. Since we could add more metrics, just unify them and delete the `SlowOpTracker`.

**Review by wenhaocs — DISMISSED** [review]

File: src/kvstore/NebulaSnapshotManager.cpp [hunk 3]
```diff
@@ -29,40 +29,69 @@ NebulaSnapshotManager::NebulaSnapshotManager(NebulaStore* kv) : store_(kv) {
...
+        "Cannot fetch the commit log id and term of space {} part {}", spaceId, partId);
+    cb(kInvalidLogId, kInvalidLogTerm, data, totalCount, totalSize, raftex::SnapshotStatus::FAILED);
+    return;
+  }
+  CHECK_EQ(val.size(), sizeof(LogID) + sizeof(TermID));
+  LogID commitLogId;
+  TermID commitLogTerm;
+  memcpy(reinterpret_cast<void*>(&commitLogId), val.data(), sizeof(LogID));
+  memcpy(reinterpret_cast<void*>(&commitLogTerm), val.data() + sizeof(LogID), sizeof(TermID));
+
```
Comment (kikimo) [inline src/kvstore/NebulaSnapshotManager.cpp:66]: Is it possible that the `commitLogId` and `commitLogTerm` we fetch from `part->engine()` laged behind the snapshot we took elier at line 45?

Comment (critical27) [reply src/kvstore/NebulaSnapshotManager.cpp:66]: Good catch! I forget to push a commit~
This PR is meant to fix this issue. We can use the rocksdb snapshot from line 45, so everything is consistant.

**Review by wenhaocs — APPROVED** [review]

File: src/kvstore/NebulaSnapshotManager.cpp [hunk 4]
```diff
@@ -29,40 +29,69 @@ NebulaSnapshotManager::NebulaSnapshotManager(NebulaStore* kv) : store_(kv) {
...
+  auto snapshot = store_->GetSnapshot(spaceId, partId);
   SCOPE_EXIT {
     if (snapshot != nullptr) {
       store_->ReleaseSnapshot(spaceId, partId, snapshot);
     }
   };
+  auto part = nebula::value(partRet);
+  // Get the commit log id and commit log term of specified partition
+  std::string val;
+  auto commitRet = part->engine()->get(NebulaKeyUtils::systemCommitKey(partId), &val, snapshot);
```
Comment (panda-sheep) [inline src/kvstore/NebulaSnapshotManager.cpp:54]: 👍
The snapshot data sent is consistent with the commitLogId and commitLogTerm in the snapshot.

**Review by wenhaocs — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]
Good job！

**Pull request merged**